homework-submission/src/App.js

```diff
@@ -0,0 +1,99 @@
+
+import React, { Component } from 'react';
+import logo from './logo.svg';
+import './App.css';
+import { observer,inject}  from 'mobx-react';
+import MyList from './myList';
+import AddTodo from './addTodos';
+import Edit from './edit'
+@inject("StoreTodo")
+@observer
+class App extends Component {
+//state for editing purpose only
+  state={
+    description:'',
+    deadline:'',
+    index:'',
+    showEdit:false
+  }
+  onChange=(index,event)=>{
+    return this.props.StoreTodo.onChange(index,event)
+  }
+  handleDeadlineChange=(value)=>{
+   return this.props.StoreTodo.handleDeadlineChange(value)
+  }
+  handleDescriptionChange=(value)=>{
+   return this.props.StoreTodo.handleDescriptionChange(value)
+  }
+  handlesubmit=(e)=>{
+   e.preventDefault();
+   let description=this.props.StoreTodo.todos.description;
+   let deadline=this.props.StoreTodo.todos.deadline;
+   let mydata={description:description,deadline:deadline}
+   return this.props.StoreTodo.handlesubmit(mydata)
+  }
+  onDelete=(index)=>{
+    this.props.StoreTodo.todoCount
+   return this.props.StoreTodo.onDelete(index)
+  }
+  onEdit=async(index,description,deadline)=>{
```

Comment:
Why **async**??

Reply:
where is it? i can't see this in the homework4.check it please.

Reply:
I was reviewing this branch `gide-homework4`. Is it the correct one?

Reply:
i supposed i have deleted the  async  on onEdit at all, unfortunately i havent pushed afterwards, but its obviously clear i dont need to use async no talking about async in this homework at all, so its mistakly done as i have tried a lot  to make the homework working then when once it came to work i forget to clean up my code, i dont thing u guys bother this much , i though this homework is about , if the homework works or not.